src/components/CommonForm/index.js

```diff
@@ -44,14 +40,44 @@ const CommonForm = ({ form, title, submit_title, submit_body }) => {
       behavior: "smooth"
     });
   };
-
+  // Validating role
+  const isValidRole = (value) => {
+    if (!value) {
+      setValidateRole(true);
+      return false;
+    } else {
+      setValidateRole(false);
+      return true;
+    }
+  };
+  /*
+  Validating Email ----
+  1. Check for email if it satisfies the standard forn for email
+  2. Check for anonymous email address/domains. Currently It is checking for only 4 domains, we can add more to regex expression as we move ahead.
+  */
+  const isValidEmail = (value) => {
+    const validEmail = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/.test(value);
```

Comment:
Also, the email value is already checked whether it is valid is not via the HTML tag type="email" so adding this additional regex to check the email wouldn't be necessary I felt.

Reply:
You are totally right it wasn't needed but we can say it just adds another layer of assurance that people will correct email address. Also if we go by the regex given in pattern attribute sagar@singh.com5 it will be an invalid email address since there is 5, so the additional check will detect this as well.